src/openrct2/drawing/Drawing.Sprite.cpp

```diff
@@ -282,6 +282,20 @@ bool gfx_load_g2()
         // Read element data
         _g2.data = fs.ReadArray<uint8_t>(_g2.header.total_size);
 
+        if (_g2.header.num_entries != G2_SPRITE_COUNT)
+        {
+            std::string errorMessage = "Mismatched g2.dat size.\nExpected: " + std::to_string(G2_SPRITE_COUNT) + "\nActual: "
+                + std::to_string(_g2.header.num_entries) + "\ng2.dat may be installed imporperly.\nPath to g2.dat: " + path;
+
+            log_error(errorMessage.c_str());
+
+            if (!gOpenRCT2Headless)
```

Comment:
I wonder if it is necessary, or if the same condition can be expressed with `uiContext != nullptr`? @IntelOrca